### PR TITLE
move nftables creation inside the controller constructor

### PR DESCRIPTION
**Pull request by aojea**
/assign @danwinship

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -150,6 +143,9 @@ func main() {
 		banpInformer,
 		cfg,
 	)
+	if err != nil {
+		panic(err.Error())
+	}
```
Comment (danwinship) [inline cmd/main.go:148]: Why panic when it was `klog.Fatalf` before?
Semantically, this doesn't seem panic-worthy

File: pkg/networkpolicy/controller_test.go [hunk 2]
```diff
@@ -92,27 +101,201 @@ func newController() *networkpolicyController {
 	npaClient := npaclientfake.NewSimpleClientset()
 	npaInformerFactory := npainformers.NewSharedInformerFactory(npaClient, 0)
 
-	controller := NewController(client,
-		&knftables.Fake{},
```
Comment (danwinship) [inline pkg/networkpolicy/controller_test.go:96]: Ah, that's the answer to "why is the knftables.Interface an argument".
(FTR, `&knftables.Fake{}` is wrong; you should use `knftables.NewFake()`)

File: pkg/networkpolicy/controller_test.go [hunk 3]
```diff
@@ -92,27 +101,201 @@ func newController() *networkpolicyController {
...
+	}
+
+	networkpolicyInformer := informersFactory.Networking().V1().NetworkPolicies()
+	namespaceInformer := informersFactory.Core().V1().Namespaces()
+	podInformer := informersFactory.Core().V1().Pods()
+	nodeInformer := informersFactory.Core().V1().Nodes()
+	adminNetworkPolicyInformer := npaInformerFactory.Policy().V1alpha1().AdminNetworkPolicies()
+
+	baselineAdminNetworkPolicyInformer := npaInformerFactory.Policy().V1alpha1().BaselineAdminNetworkPolicies()
+	c := &Controller{
```
Comment (danwinship) [inline pkg/networkpolicy/controller_test.go:116]: blank line in the wrong place?

File: pkg/networkpolicy/controller_test.go [hunk 4]
```diff
@@ -92,27 +101,201 @@ func newController() *networkpolicyController {
...
+			if pod.Spec.HostNetwork {
+				return []string{}, nil
+			}
+			result := []string{}
+			for _, ip := range pod.Status.PodIPs {
+				result = append(result, string(ip.IP))
+			}
+			return result, nil
+		},
+	})
```
Comment (danwinship) [inline pkg/networkpolicy/controller_test.go:141]: it seems crazy to duplicate all of this in the unit test; you should just have a `newControllerInternal()` that takes the knftables arg.

or call `NewController()` and just overwrite `c.nft` afterward

File: cmd/main.go (hunk 1 above)
Comment (aojea) [reply cmd/main.go:148]: yeah it has to exit with error because the controller is not able to start

**Pull request merged**